.github/workflows/build.yml

```diff
@@ -2,6 +2,79 @@ name: CI
 on: [push]
 
 jobs:
+    licenses:
+        name: Check licenses
+        runs-on: ubuntu-latest
+
+        steps:
+          - uses: actions/setup-node@v3
+          - name: Checkout sources
+            uses: actions/checkout@v2
+            with:
+                persist-credentials: false
+
+          - name: Check License Header (Fix Mode)
+            uses: apache/skywalking-eyes/header@v0.5.0
+            with:
+              mode : fix
+
+          - name: Get current Pull Request number
+            uses: actions/github-script@v7
+            id: get_issue_number
+            with:
+              script: |
+                if (context.issue.number) {
+                  // Return issue number if present
+                  return context.issue.number;
+                } else {
+                  // Otherwise return issue number from commit
+                  return (
+                    await github.rest.repos.listPullRequestsAssociatedWithCommit({
+                      commit_sha: context.sha,
+                      owner: context.repo.owner,
+                      repo: context.repo.repo,
+                    })
+                  ).data[0].number;
+                }
+              result-encoding: string
+
+          - name: Create Pull Request if necessary
+            if: github.ref != 'refs/heads/main' && ${{ steps.get_issue_number.outputs.result }} != ''
```

Comment:
create a new PR _if_ it's different of `main` branch? isn't the inverse?

Reply:
nope, we don't want this step of this job to be executed at push event on main. This way it will create a PR only on push event on a branch with an existing PR.